cmd/statusd/debug/debug.go

```diff
@@ -0,0 +1,104 @@
+package debug
+
+import (
+	"bufio"
+	"fmt"
+	"log"
+	"net"
+	"reflect"
+
+	"github.com/status-im/status-go/geth/api"
+)
+
+// Debug provides a server receiving line based commands from a
+// CLI via the debugging port and executing those on the Status API
+// using reflection. The returned values will be rendered as
+// string and returned to the CLI.
+type Debug struct {
+	csv      reflect.Value
+	listener net.Listener
+}
+
+// New creates a debugger using the oassed Status API.
+// It also starts the server.
+func New(statusAPI *api.StatusAPI) (*Debug, error) {
+	listener, err := net.Listen("tcp", ":51515") // nolint
+	if err != nil {
+		return nil, err
+	}
+	d := &Debug{
```

Comment:
In such a case, it's usually a good pattern to get a struct's pointer in `return` statement like `return &d, nil`.

Reply:
More often seen it the other way so far, but don't care.

Reply:
Yeah, me neither in this case. But there is an interesting blog post about it. Fragment from https://www.goinggo.net/2017/05/language-mechanics-on-escape-analysis.html:

> If you do this, you are walking away from an important readability gain you can have in your code. Step away from the entire function for a second and just focus on the return.
Listing 6
34     return u
35 }
What does this return tell you? All that it says is that a copy of u is being passed up the call stack. However, what does the return tell you when you use the & operator?
Listing 7
34     return &u
35 }
Thanks to the & operator, the return now tells you that u is being shared up the call stack and therefore escaping to the heap. Remember, pointers are for sharing and replace the & operator for the word “sharing” as you read code. This is very powerful in terms of readability, something you don’t want to lose.